presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -446,6 +449,16 @@ public static NullableValue parsePartitionValue(String partitionName, String val
             return NullableValue.of(TIMESTAMP, timestampPartitionKey(value, timeZone, partitionName));
         }
 
+        if (REAL.equals(type)) {
+            if (isNull) {
+                return NullableValue.asNull(REAL);
+            }
+            if (value.isEmpty()) {
+                return NullableValue.of(REAL, floatToRawIntBits(0.0f));
```

Comment:
Good catch but why c-tor of `NullableValue` is not yelling at me if there's a check:
```java
!Primitives.wrap(type.getJavaType()).isInstance(value)) {
throw new IllegalArgumentException(String.format("Object '%s' does not match type %s", value, type.getJavaType()));
```
`javaType` of `REAL` type is `long`.
Maybe we don't have tests covering such a case?
BTW I just repeated schema from other types but can you explain to me when partition value may become empty (size() = 0)? Shouldn't it always be a pair of `key=value`? Is it legal to have `key=` or just `key` in Hive?

Anyway, fixed in 623e2b8411bcd16eacf1e93ee4f466927fdb2a63.